src/app/Library/CrudPanel/Traits/FieldsProtectedMethods.php

```diff
@@ -138,14 +138,15 @@ protected function makeSureFieldHasEntity($field)
 
         //if the name is dot notation we are sure it's a relationship
         if (strpos($field['name'], '.') !== false) {
-            $field['entity'] = $field['name'];
-
-            return $field;
+            $possibleMethodName = Str::endsWith(Str::before($field['name'], '.'), '_id') ? Str::replaceLast('_id', '', Str::before($field['name'], '.')) : Str::before($field['name'], '.');
```

Comment:
Also, I think refactoring this to fluent string helpers will also help legibility, in general I find the fluent string helpers MUUUCH easier to read than the normal ones, I think they're a HUGE benefit of us requiring only Laravel 8+... we can use them everywhere now ❤️❤️❤️

```php
// this
Str::of($field['name'])->before('.')->endsWith('_id')
// is much easier to read than this
Str::endsWith(Str::before($field['name'], '.'), '_id')
```

After that, _I belive_ if you run a replaceLast on a string and it doesn't find anything, it will still return the intact string. So the conditional might be for nothing there.

It's possible the whole conditional above could become:

```php
$firstSegment = Str::of($field['name'])->before('.');
$possibleMethodName = $firstSegment->replaceLast('_id'); // just in case

// or maybe even
$possibleMethodName = Str::of($field['name'])->before('.')->replaceLast('_id'); // first segment without "_id"
```

Is this what that's meant to do?

Reply:
Also, question... if that's what it does. Why would anybody write `product_id.category.name`? Should we even support that?! Is the last `replaceLast()` really needed here?

Reply:
Thanks for the review @tabacitu 

That was there for another usecase, that is using the attribute in the relation string.

So you would setup ```'name' => 'belongsTo.attribute'```php meaning you could aswell setup ```'name' => 'belongsTo_id.attribute'```php . 

I've refactored and removed it from here. We will be discussing if we should support this later (there is more code needed to account for this scenario) I just let it here since it does not impact with other scenarios, was specific for this one. But better introduce everything in the same PR.

Reply:
Ok sure. Glad my spidey sense still works sometimes 😅 I didn't quite understand from the above, but... different problem, different discussion. Thanks!